### PR TITLE
Add insights dashboard view for A/B testing blocks

### PR DESCRIPTION
**Pull request by roborourke**
Fixes https://github.com/humanmade/altis-analytics/issues/191

### CODE REVIEW

File: inc/utils/namespace.php [hunk 1]
```diff
@@ -871,3 +871,13 @@ function get_countries() : array {
...
 }
+
+/**
+ * Get a letter of the alphabet corresponding to the passed zero based index.
+ *
+ * @param integer $index Letter of the alphabet to get.
+ * @return string
+ */
+function get_letter( int $index ) : string {
+	return strtoupper( 'abcdefghijklmnopqrstuvwxyz'[ $index ] ?? '' );
```
Comment (shadyvb) [inline inc/utils/namespace.php:882]: #morningfun #beforecoffee 

```php
function get_letter( int $index ) : string {
    $letters = $index / 26;
    return str_repeat( 'z', floor( $letters ) ) . ( fmod( $letters, 1 ) ? chr( ord( 'a' ) + 26 * fmod( $letters, 1 ) - 1 ) : '' );
}

var_dump( implode( ",", array_map( 'get_letter', range(1, 100) ) ) );

// "a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,za,zb,zc,zd,ze,zf,zg,zh,zi,zj,zk,zl,zm,zn,zo,zp,zq,zr,zs,zt,zu,zv,zw,zx,zy,zz,zza,zzb,zzc,zzd,zze,zzf,zzg,zzh,zzi,zzj,zzk,zzl,zzm,zzn,zzo,zzp,zzq,zzr,zzs,zzt,zzu,zzv,zzw,zzx,zzy,zzz,zzza,zzzb,zzzc,zzzd,zzze,zzzf,zzzg,zzzh,zzzi,zzzj,zzzk,zzzl,zzzm,zzzn,zzzo,zzzp,zzzq,zzzr,zzzs,zzzt,zzzu,zzzv"
```

Woke up as I started to optimize for `zaa, zab, ...` 😄 (will do so later 😛)

File: src/blocks/ui/components/block-ab-test.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+/**
+ * Personalized Content Block Analytics component.
+ *
+ * @param {object} props The component props.
+ * @param {object} props.block The block post data.
+ * @param {string} props.clientId The block client ID.
+ * @returns {React.ReactNode} The block view component.
+ */
+const BlockABTest = ( {
```
Comment (shadyvb) [inline src/blocks/ui/components/block-ab-test.js:29]: Does it make sense to namespace this somehow ?

File: src/blocks/ui/components/variants.js [hunk 3]
```diff
@@ -85,17 +28,16 @@ function Variants( props ) {
...
 		<VariantsList>
-			<h2>{ __( 'Block Variants', 'altis-analytics' ) }</h2>
 			{ variants.map( variant => {
-				const data = ( analytics && analytics.audiences.find( audience => audience.id === ( variant.audience && variant.audience.id ) || 0 ) ) || defaultVariantAnalytics;
+				const data = ( analytics && analytics.variants.find( v => v.id === variant.id || 0 ) ) || defaultVariantAnalytics;
 				return (
 					<div className={ `altis-analytics-block-variant ${ variant.fallback ? 'altis-analytics-block-variant--fallback' : '' }` }>
 						<div className="altis-analytics-block-variant__header">
-							<h3>{ variant.fallback ? __( 'Fallback', 'altis-analytics' ) : decodeEntities( variant.audience.title ) }</h3>
+							<h3>{ ! variant.title && variant.fallback ? __( 'Fallback', 'altis-analytics' ) : decodeEntities( variant.title ) }</h3>
```
Comment (shadyvb) [inline src/blocks/ui/components/variants.js:40]: Better to account for translated content as well.

```suggestion
							<h3>{ decodeEntities( ! variant.title && variant.fallback ? __( 'Fallback', 'altis-analytics' ) : variant.title ) }</h3>
```

**Review by shadyvb — APPROVED** [review]
Looks good, nice work!

File: src/blocks/ui/components/variants.js (hunk 3 above)
Comment (roborourke) [reply src/blocks/ui/components/variants.js:40]: Ah nice catch!

File: src/blocks/ui/components/block-ab-test.js (hunk 2 above)
Comment (roborourke) [reply src/blocks/ui/components/block-ab-test.js:29]: Yeah it's only really namespaced by the directory it's in right now 🤔  I'll have a think.

File: inc/utils/namespace.php (hunk 1 above)
Comment (roborourke) [reply inc/utils/namespace.php:882]: oh man, not done any code katas in ages 😂 

This a zero based index so 0 should equal "a". I'd probably go for a-z then aa, ab, ac, ... ba, bb, bc etc... Might see if I can figure that out.

Comment (shadyvb) [reply inc/utils/namespace.php:882]: Well, definitely not needed here :D i just did some work with `ord` and `chr` before so thought I'd try if it fits.

**Pull request merged**